sky/cli.py

```diff
@@ -1607,10 +1607,33 @@ def logs(
 def cancel(cluster: str, all: bool, jobs: List[int]):  # pylint: disable=redefined-builtin
     # NOTE(dev): Keep the docstring consistent between the Python API and CLI.
     """Cancel job(s)."""
+    bold = colorama.Style.BRIGHT
+    reset = colorama.Style.RESET_ALL
+    if not jobs and not all:
+        # Friendly message for usage like 'sky cancel 1'.
+        raise click.UsageError(
+            f'Use {bold}sky cancel <cluster> <job ID>{reset} '
+            f'or {bold}sky cancel <cluster> --all{reset} to cancel one '
+            'or all jobs on a cluster. Job IDs can be looked up by '
+            f'{bold}sky queue{reset}.')
     try:
         core.cancel(cluster, all, jobs)
     except ValueError as e:
-        raise click.UsageError(str(e))
+        if 'Cancelling jobs is not allowed' in str(e):
+            # Friendly message for usage like 'sky cancel <spot controller>
+            # -a/<job id>'.
+            if all:
+                arg_str = '--all'
+            else:
+                arg_str = ' '.join(map(str, jobs))
+            error_str = (
+                'Cancelling the spot controller\'s jobs is not allowed.'
+                f'\nTo cancel spot jobs, use: {bold}sky spot cancel <spot '
+                f'job IDs> [--all]{reset}'
+                f'\nDo you mean: {bold}sky spot cancel {arg_str}{reset}')
+        else:
+            error_str = str(e)
+        raise click.UsageError(error_str)
```

Comment:
Nice! Is it cleaner to use a dedicated exception for the case when the user is trying to cancel the job on the spot controller, instead of using the `ValueError`?
Reason: checking the string in the exception can be less robust, as we need to keep the string value checking and the error message in sync, say if we change the error message in the future, but forget to change it here, it will fail.

Reply:
Good call. Done.